distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date_Time.enso

```diff
@@ -380,18 +379,24 @@ type Date_Time
     ## Add the specified amount of time to this instant to produce a new instant.
 
        Arguments:
-       - amount: The amount of time to add to this instant.
+       - amount: The amount of time to add to this instant, either `Duration` for
+         adding time (hours, minutes, etc.), or `Period` for adding date (days,
+         months, years).
 
        > Example
          Add 15 years and 3 hours to a zoned date time.
 
              from Standard.Base import Date_Time
-             import Standard.Base.Data.Time.Duration
+             from Standard.Base.Data.Time import Duration, Period
 
-             example_plus = Date_Time.new 2020 + 15.years + (Duration.new hours=3)
-    + : Duration -> Date_Time ! Time_Error
+             example_plus = Date_Time.new 2020 + (Period.new years=15) + (Duration.new hours=3)
+    + : (Duration | Period) -> Date_Time ! Time_Error
     + self amount =
-        Panic.catch ArithmeticException (self.plus_builtin amount) (err -> Error.throw (Time_Error_Data err.getMessage))
+        case amount of
+            duration : Duration.Duration ->
+                Panic.catch ArithmeticException (self.plus_builtin duration) (err -> Error.throw (Time_Error_Data err.getMessage))
+            period : Period.Period ->
```

Comment:
Right. `duration` is handled as built-in but `period` is handled via `Time_Utils`.